specification/servicenetworking/resource-manager/Microsoft.ServiceNetworking/preview/2024-05-01-preview/TrafficController.json

```diff
@@ -0,0 +1,1895 @@
+      "type": "object",
+      "description": "The response of a TrafficController list operation.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The TrafficController items on this page",
+          "items": {
+            "$ref": "#/definitions/TrafficController"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "TrafficControllerProperties": {
+      "type": "object",
+      "description": "Traffic Controller Properties.",
+      "properties": {
+        "configurationEndpoints": {
+          "type": "array",
+          "description": "Configuration Endpoints.",
+          "items": {
+            "type": "string"
+          },
+          "readOnly": true
+        },
+        "frontends": {
+          "type": "array",
+          "description": "Frontends References List",
+          "items": {
+            "$ref": "#/definitions/ResourceId"
+          },
+          "readOnly": true
+        },
+        "associations": {
+          "type": "array",
+          "description": "Associations References List",
+          "items": {
+            "$ref": "#/definitions/ResourceId"
+          },
+          "readOnly": true
+        },
+        "securityPolicies": {
+          "type": "array",
+          "description": "Security Policies References List ",
+          "items": {
+            "$ref": "#/definitions/ResourceId"
+          },
+          "readOnly": true
+        },
+        "securityPolicyConfigurations": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SecurityPolicyConfigurations"
+          },
+          "readOnly": true,
+          "x-ms-identifiers": []
+        },
+        "provisioningState": {
+          "$ref": "#/definitions/ProvisioningState",
+          "description": "The status of the last operation.",
+          "readOnly": true
+        }
+      }
+    },
+    "TrafficControllerUpdate": {
+      "type": "object",
+      "description": "The type used for update operations of the TrafficController.",
+      "properties": {
+        "tags": {
+          "type": "object",
+          "description": "Resource tags.",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "Versions": {
+      "type": "string",
+      "description": "Api versions",
+      "enum": [
+        "2024-05-01-preview"
+      ],
+      "x-ms-enum": {
+        "name": "Versions",
+        "modelAsString": true,
+        "values": [
+          {
+            "name": "2024-05-01-preview",
+            "value": "2024-05-01-preview",
+            "description": "2024-05-01 preview version"
+          }
+        ]
+      }
+    },
+    "WAFPolicy": {
+      "type": "object",
+      "properties": {
+        "id": {
```

Comment:
>"id" [](http://example.com/codeflow?start=8&length=4)

Can this id (and across the page) be referring to ResourceId definition ? had have the checks/validations inside ResourceId if needed?

Reply:
Right now it is the same as the ResourceId object but we are defining different types to allow them to be extensible in the future. We expect different properties to be added to each policy.